### ISSUE TITLE
Renaming Component Guide Sections and Introducing Core Concepts

### ISSUE THREAD

**Issue by pzuraq** [ember-learn/guides-source#767]
Currently, the Component guide sections are named in a _semi-narrative_ manner, with each section following the previous section in order, and section names primarily trying to describe the _functionality_ that they are going to teach.

```
Defining a Component
The Component Lifecycle
Passing Properties to a Component
Wrapping Content in a Component
Customizing a Component's Element
Using Block Params
Handling Events
Triggering Changes with Actions
```

This naming is good for newcomers and beginners who are learning for Ember for the first time, since it reads like a tutorial, building up knowledge section over section, and is descriptive of the functionality, so it makes it easier to find a section based on the functionality that the user is trying to use. However, it is problematic for _existing_ Ember users, or users who are already familiar with Ember's terminology, since there is no clear way to connect the descriptions to the concepts they are describing at a glance. This is especially confusing when certain concepts, such as `{{yield}}` and blocks are broken up into multiple sections (`Wrapping Content in a Component` and `Using Block Params`) in order to fit the narrative structure.

In the Octane guides, we've changed the names of these sections to match up with the _**core, architectural concepts**_  that are being taught. The reasoning behind this is:

1. These concepts and their terms are fundamental to Ember, and changing them would represent a massive shift in the mental model, so we know they will be stable.
2. Listing these core concepts early on gives newcomers an overview of the important architectural concepts that they should associate with components in Ember, and crystalizes the core concepts and their names for the community as a whole.
3. Using the terms benefits return Ember users, and instructors/community members who are trying to help new users out, since they can quickly refer to the guide by the term that they commonly use.

The proposed new Component guides sections are:

```
Component Basics
Defining a Component
Arguments and Attributes
Actions and Events
Slotting with Yields
Contextual Components
Working with the DOM
```

We also use standard HTML terminology here to associate as much as we can with related/standardized terms, such as HTML attributes (which are discussed in depth, and highly related to _arguments_), and [slots](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot), which are similar enough to `{{yields}}` that associating them will likely help new users and SEO.

While this change improves the user story for existing Ember developers, we don't want it to _detract_ from the experience for new users. We can make sure that user story is addressed in two ways:

* We can focus on improving the learning story in _tutorials_ and the _quickstart_, where a narrative/descriptive structure continues to make sense, and where new users will likely start off. 
* We can add a _terminology cheatsheet_, which defines Ember terms like "yields" and "component arguments" in a quick and descriptive manner, and links to the appropriate guides.
* For users of _other_ frameworks, we should invest in quickstart guides specifically focusing on the framework, so we can show them very quickly 1-1 examples that can help them to be productive. We can also create terminology cheatsheets specific to each framework, since they may have their own terms (e.g. Props in React -> Arguments in Ember)

Between these resources, users who are unfamiliar with Ember's terminology will have quick ways to find the core terms that they are looking for, and their guides pages.

**Comment by pzuraq** [comment]
One quick note here is that I updated the name for yields to `Slotting with Yields`, some other ideas I had were `Slots and Yields` or `Slots with Yields`, but I think that implies that we _have_ slots as a concept in Ember, when we don't. What we want to imply is that our _equivalent_ to slots is yields, ideally with both terms in the title so we connect the core concept.

Slots also are not _fully_ standard yet (they don't ship in IE/Edge) so I don't think we should jump the gun on trying to _replace_ our core concept with them yet.

**Comment by jenweber** [comment]
Closed by #1150
